File: festival-web/src/main/java/cn/bdqfork/web/route/message/resolver/ParameterResolver.java
package cn.bdqfork.web.route.message.resolver;

import io.vertx.reactivex.ext.web.RoutingContext;

import java.lang.reflect.Parameter;

/**
 * @author bdq
 * @since 2020/2/11
 */
public interface ParameterResolver {
    Object resolve(Parameter parameter, RoutingContext routingContext);
}

File: festival-kotlin-web/src/main/java/cn/bdqfork/web/route/message/resolver/ParameterResolver.kt
package cn.bdqfork.web.route.message.resolver

import io.vertx.ext.web.RoutingContext
import java.lang.reflect.Parameter

/**
 * @author bdq
 * @since 2020/2/11
 */
interface ParameterResolver {
    fun resolve(parameter: Parameter, routingContext: RoutingContext): Any?
}